engine/src/main/java/org/hibernate/search/engine/cfg/spi/ParseUtils.java

```diff
@@ -400,4 +400,89 @@ private static String[] arrayValues(String value) {
 		}
 		return value.split( "[,;\\s]+" );
 	}
+
+	public static String format(BigDecimal value) {
+		return value.toPlainString();
+	}
+
+	public static String format(BigInteger value) {
+		return value.toString();
+	}
+
+	public static String format(Boolean value) {
+		return value.toString();
+	}
+
+	public static String format(Byte value) {
+		return value.toString();
+	}
+
+	public static String format(Double value) {
+		return value.toString();
+	}
+
+	public static String format(Float value) {
+		return value.toString();
+	}
+
+	public static String format(GeoPoint value) {
+		// to match parseGeoPoint
+		return String.format( Locale.ROOT, "%s, %s", value.latitude(), value.longitude() );
+	}
+
+	public static String format(Instant value) {
+		return DateTimeFormatter.ISO_INSTANT.format( value );
+	}
+
+	public static String format(Integer value) {
+		return value.toString();
+	}
+
+	public static String format(LocalDate value) {
+		return DateTimeFormatter.ISO_LOCAL_DATE.format( value );
+	}
+
+	public static String format(LocalDateTime value) {
+		return DateTimeFormatter.ISO_LOCAL_DATE_TIME.format( value );
+	}
+
+	public static String format(LocalTime value) {
+		return DateTimeFormatter.ISO_LOCAL_TIME.format( value );
+	}
+
+	public static String format(Long value) {
+		return value.toString();
+	}
+
+	public static String format(MonthDay value) {
+		return ISO_MONTH_DAY.format( value );
+	}
+
+	public static String format(OffsetDateTime value) {
+		return DateTimeFormatter.ISO_OFFSET_DATE_TIME.format( value );
+	}
+
+	public static String format(OffsetTime value) {
+		return DateTimeFormatter.ISO_OFFSET_TIME.format( value );
+	}
+
+	public static String format(Short value) {
+		return value.toString();
+	}
+
+	public static String format(String value) {
+		return value;
+	}
+
+	public static String format(Year value) {
+		return ISO_YEAR.format( value );
+	}
+
+	public static String format(YearMonth value) {
+		return ISO_YEAR_MONTH.format( value );
+	}
+
+	public static String format(ZonedDateTime value) {
+		return DateTimeFormatter.ISO_ZONED_DATE_TIME.format( value );
+	}
```

Comment:
Shouldn't this be in `FormatUtils`? Maybe not, IDK.

In any case, this probably warrants some tests, e.g. in `org.hibernate.search.engine.cfg.spi.ParseUtilsTest`.

Reply:
added `org.hibernate.search.engine.cfg.spi.FormatUtils`